### DISCUSSION TITLE
Idea Generation Agent (Optimization)

### DISCUSSION THREAD

**Discussion by BradKML** [SmartManoj/Kevin#143] [General]
Borrowed from this issue of AI Scientist https://github.com/SakanaAI/AI-Scientist/issues/19
Paper idea generation tools (see also https://github.com/Paureel/LLM-SCI-GEN for more papers)
- https://github.com/EagleW/PaperRobot
- https://github.com/cheerss/SciPIP
- https://github.com/artificial-scientist-lab/SciMuse

Something worth considering is how they parse papers differently than just dumping them in as a pure article.

**Comment by SmartManoj** [comment]
Should it be another agent like `CodeActAgent`? or Should `CodeActAgent` use these tools?

**Reply by BradKML** [reply]
For some reason having another agents would be best when it comes to creative thinking, like internal dialogues but also have stronger preference for RAG work (e.g. having a list of papers or articles provided).

**Reply by SmartManoj** [reply]
You could change the [system prompt](https://github.com/All-Hands-AI/OpenHands/blob/main/openhands/agenthub/codeact_agent/prompts/system_prompt.j2) for creative thinking.

Why should RAG be used here? Shouldn't the LLM read the Full paper?

[Clean long PDFs for LLM inference](https://console.upstage.ai/docs/getting-started/tutorials/clean-pdf)

**Reply by BradKML** [reply]
Thinking RAG is needed if the library has more than a full list of more than a dozen writers, each having like more than a dozen books and papers, so imagine >100 papers blowing up the context window. In the case of pure idea generation rather than topic-centric generation, most people still have >100 papers but spread across a larger set of writers.

**Reply by SmartManoj** [reply]
Still, should the agent process all the PDFs fully or only a subset of it?

**Reply by BradKML** [reply]
All of the PDFs assuming the selection is bespoke, but a subset of information (or a summary of information) for each of the PDFs. RAG and/or CoD would be needed.
